madmom/evaluation/chords.py

```diff
@@ -0,0 +1,969 @@
+
+
+def adjust(det_chords, ann_chords):
+    """
+    Adjust the length of detected chord segments to the annotation
+    length.
+
+    Discard detected chords that start after the annotation ended,
+    and shorten the last detection to fit the last annotation;
+    discared detected chords that end before the annotation begins,
+    and shorten the first detection to match the first annotation.
+
+    Parameters
+    ----------
+    det_chords : numpy structured array
+        Detected chord segments.
+    ann_chords : numpy structured array
+        Annotated chord segments.
+
+    Returns
+    -------
+    det_chords : numpy structured array
+        Adjusted detected chord segments.
+    """
+    det_start = det_chords[0]['start']
+    ann_start = ann_chords[0]['start']
+    if det_start > ann_start:
+        filler = np.array((ann_start, det_start, chord('N')),
+                          dtype=CHORD_ANN_DTYPE)
+        det_chords = np.hstack([filler, det_chords])
+    elif det_start < ann_start:
+        det_chords = det_chords[det_chords['end'] > ann_start]
+        det_chords[0]['start'] = ann_start
+
+    det_end = det_chords[-1]['end']
+    ann_end = ann_chords[-1]['end']
+    if det_end < ann_end:
+        filler = np.array((det_end, ann_end, chord('N')),
+                          dtype=CHORD_ANN_DTYPE)
+        det_chords = np.hstack([det_chords, filler])
+    elif det_end > ann_end:
+        det_chords = det_chords[det_chords['start'] < ann_end]
+        det_chords[-1]['end'] = ann_chords[-1]['end']
+
+    return det_chords
+
+
+def segmentation(ann_starts, ann_ends, det_starts, det_ends):
+    """
+    Compute the normalized Hamming divergence between chord
+    segmentations as defined in [1]_ (Eqs. 8.37 and 8.38).
+
+    Parameters
+    ----------
+    ann_starts : list or numpy array
+        Start times of annotated chord segments.
+    ann_ends : list or numpy array
+        End times of annotated chord segments.
+    det_starts : list or numpy array
+        Start times of detected chord segments.
+    det_ends : list or numpy array
+        End times of detected chord segments.
+
+    Returns
+    -------
+    distance : float
+        Normalised Hamming divergence between annotated and
+        detected chord segments.
+
+    References
+    ----------
+    .. [1] Christopher Harte, "Towards Automatic Extraction of Harmony
+           Information from Music Signals." Dissertation,
+           Department for Electronic Engineering, Queen Mary University of
+           London, 2010.
+    """
+    est_ts = np.unique(np.hstack([det_starts, det_ends]))
+    seg = 0.
+    for start, end in zip(ann_starts, ann_ends):
+        dur = end - start
+        seg_ts = np.hstack([
+            start, est_ts[(est_ts > start) & (est_ts < end)], end])
+        seg += dur - np.diff(seg_ts).max()
+
+    return seg / (ann_ends[-1] - ann_starts[0])
+
+
+class ChordEvaluation(object):
```

Comment:
Actually, I'm not sure what this MixIn is for:
 - Every evaluation implements its own `to_string` method.
 - The `__len__` method is not used anywhere (in any case, it can only return an integer, so it is quite limited)
 - The `metrics` property is not used anywhere except for the `AlignmentEvaluation`
 - The formatters `tocsv` and `totex` only use `METRIC_NAMES`, which has to be defined for each evaluation class anyways.

So, what advantages does the Mixin give us?